connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -548,6 +548,7 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
                         // Connector deletion will be written as a null value
                         log.info("Successfully processed removal of connector '{}'", connectorName);
                         connectorConfigs.remove(connectorName);
+                        taskConfigs.keySet().removeIf(taskId -> taskId.connector().equals(connectorName));
```

Comment:
Do you think it is worth also updating `connectorTaskCounts` to keep these in sync?

Reply:
No kidding! Good catch. I've added logic in this class to do that, and expanded the test to verify this logic.